### PR TITLE
Added CLI version number functionality

### PR DESCRIPTION
**Pull request by Scafir**
PR for issue #487. 
Pretty simple stuff, but review it carefully, this is my first real PR ever :smiley:

### CODE REVIEW

**Comment by dreua** [comment]
First of all congratulations and thanks for your first PR ☺️

**Comment by dreua** [comment]
The CI fails for some reason that is not directly clear to me, but it's getting too late for me to investigate this now.

File: pdfarranger/pdfarranger.py
Comment (dreua) [reply pdfarranger/pdfarranger.py:276]: https://repology.org/project/glib/versions

**Comment by Scafir** [comment]
I had a look at it, and it's indeed surprising, on one hand, it seems that dogtail's `keyCombo` is broken with my modifications, and on the other, the master branch does not pass `test_05_duplicate`..

File: pdfarranger/pdfarranger.py
Comment (jeromerobert) [reply pdfarranger/pdfarranger.py:276]: Indeed we don't need that check. I don't think anyone has ever run PDF Arranger with such an old Glib. For Debian it's before 2015 and for Ubuntu the package is not even available anymore.

**Comment by jeromerobert** [comment]
This is probably a bug in the test itself. There is probably a lack of synchronization and a race condition. I had lot of issues with that when writing those tests. As it looks like totally unrelated to your PR I suggest to ignore that failure for now. I'll have a look at the test.

File: pdfarranger/pdfarranger.py
Comment (Scafir) [reply pdfarranger/pdfarranger.py:286]: The changes that I made break that functionality. Could you give me a pointer on where in the code you implemented that so that can add it back and remove dead code? (I don't find it on my own..)

File: pdfarranger/pdfarranger.py
Comment (Scafir) [reply pdfarranger/pdfarranger.py]: That is correct, moreover we really don't need this check since we `gi.check_version('3.10.2')`. Corrected with commit c27e4332485619aa64c7ce68eed72fffceca67af.

**Comment by Scafir** [comment]
Moreover, the recognized binary name in the `--help` is `__main__.py`. I will have to correct that before it is merged.

File: pdfarranger/pdfarranger.py
Comment (dreua) [reply pdfarranger/pdfarranger.py]: https://github.com/pdfarranger/pdfarranger/blob/65b651e3e689acfefcc0b6e928990cb651d6277e/pdfarranger/pdfarranger.py#L271

Comment (dreua) [reply pdfarranger/pdfarranger.py]: > Moreover, the recognized binary name in the `--help` is `__main__.py`. I will have to correct that before it is merged.

This could also depend on how exactly PDF arranger ist launched. If you find a good solution that's fine but I wouldn't spend too much time or code on it. Hardcoding would be fine, too.

**Comment by dreua** [comment]
> Moreover, the recognized binary name in the `--help` is `__main__.py`. I will have to correct that before it is merged.

This could also depend on how exactly PDF arranger ist launched. If you find a good solution that's fine but I wouldn't spend too much time or code on it. Hardcoding would be fine, too.

File: pdfarranger/pdfarranger.py
Comment (Scafir) [reply pdfarranger/pdfarranger.py]: Fixed with f7463c32f159c777dc15211e6f6b8cb007f9d705. I also added error handling in case of non-existant files, which was kind of implemented before:  the first file was correct but no the second, no error message was displayed and the canevas appeared empty.
After looking at the manpages of evince, totem, etc of which none talk about optional files, I think it is okay as is. `[FILES]` is displayed nonetheless at the very beginning: `Usage:  __main__.py [OPTION…] [FILES]`

**Comment by Scafir** [comment]
Everything should be correct now. The most likely point where you may not accept my changes is the creation of the `bin` folder with an executable inside it, as opposed to the original automatic script creation with setuptools.

**Review by dreua — APPROVED** [review]
Looks good to me, but I don't know enough about entry points and best practices to merge it. Should be squashed in any case.

**Comment by Scafir** [comment]
I don't know much either about setuptools. I inspired myself from `lutris` structure, so it _should_ not be too bad.
Concerning the squashing, is it something I should do on my end? From what can see, it is not really possible without a `--force`, which I don't think is desirable.

**Comment by dreua** [comment]
Inspiration by `lutris` sounds good to me, but maybe Jerome should have a look, he built most of the setuptools stuff.

Squashing can be done on both ends. In my opinion force-pushing to a feature branch is acceptable here since I don't expect anyone else having it checked out with active work on top. So feel free to squash it down to a few commits. In this case it would be a good idea to separate the `__main__` change from the rest just in case we need to revert that later. `git rebase -i main` is my friend for this work.

File: bin/pdfarranger
Comment (dreua) [inline bin/pdfarranger:1]: I [and Fedora](https://docs.fedoraproject.org/en-US/packaging-guidelines/#_shebang_lines) would prefer `#!/usr/bin/python3`

**Review by dreua — CHANGES_REQUESTED** [review]

File: bin/pdfarranger
Comment (Scafir) [reply bin/pdfarranger]: Done and squashed into 48b5df6100f46ca641694e71fe4f8fa7d298edb4

**Review by dreua — APPROVED** [review]

**Comment by dreua** [comment]
The Dogtail tests fail completely now, unfortunately we need to have another look at that. I also get errors for about half of them when running tests locally.

**Comment by jeromerobert** [comment]
> Inspiration by lutris sounds good to me, but maybe Jerome should have a look, he built most of the setuptools stuff.

I would prefer not too keep that change. The `__main__.py` in help only appear in dev mode so we I think we don't have to care. This also break the Windows build. Once installed `pdfarranger --help` behave as expected.

**Comment by jeromerobert** [comment]
Merged as 4b7312a436ceae3a6e43e64780dc66e5d19ed7d8

**Pull request closed without merging**
